modules/json/test/src/smithy4s/http/json/JsonCodecApiTests.scala

```diff
@@ -25,4 +25,24 @@ class JsonCodecApiTests extends FunSuite {
 
     assertEquals(encodedString, """{"a":"test"}""")
   }
+
+  test(
+    "struct codec with a required field should return a Left when the field is missing"
+  ) {
+    val schemaWithRequiredField =
+      Schema
+        .struct[String]
+        .apply(
+          Schema.string
+            .required[String]("a", identity)
+        )(identity)
+
+    val capi = codecs(HintMask.empty)
+
+    val codec = capi.compileCodec(schemaWithRequiredField)
+
+    val decoded = capi.decodeFromByteArray(codec, """{}""".getBytes())
```

Comment:
Will it return `Left` for empty or invalid JSON input?

Reply:
Yep. That might be covered by other tests... but we certainly didn't have a test for this :)